config/src/config/network_config.rs

```diff
@@ -340,58 +340,95 @@ impl Default for RateLimitConfig {
     }
 }
 
-pub type TrustedPeerSet = HashMap<PeerId, TrustedPeer>;
+pub type PeerSet = HashMap<PeerId, Peer>;
+
+// TODO: Combine with RoleType?
+/// Represents the Role that a peer plays in the network ecosystem rather than the type of node.
+/// Determines how nodes are connected to other nodes, and how discovery views them.
+///
+/// Rules for upstream nodes via Peer Role:
+///
+/// Validator -> Always upstream if not Validator else P2P
+/// PreferredUpstream -> Always upstream, overriding any other discovery
+/// ValidatorFullNode -> Always upstream for incoming connections (including other ValidatorFullNodes)
+/// Upstream -> Upstream, if no ValidatorFullNode or PreferredUpstream.  Useful for initial seed discovery
+/// Unknown -> Undiscovered peer, likely due to a non-mutually authenticated connection always downstream
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+pub enum PeerRole {
+    Validator = 0,
+    PreferredUpstream,
+    ValidatorFullNode,
+    Upstream,
+    Unknown,
+}
```

Comment:
it still seems a bit weird to me, conflating "upstream-ness" with role type, but it's probably ok for now

Reply:
Yeah, we could separate it into two, but I'm trying to keep it simple.

This ordering specifically allows us to say our full node we want to override our preferred upstream node to say a private or specific full node we believe we trust.  The `Upstream` allows us to say we want to seed ourselves in and then if we find a better node we can use it instead.  I'm sure they could be renamed to like, Validator, Preferred, ValidatorFullNode, Backup?, and Unknown.